ministryofjustice/laa-review-criminal-legal-aid
# [CRIMAP-563] Add Data Analyst Role

## Description of change
Introduces a new Data Analyst role 

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAP-563

## Notes for reviewer
This ticket will require QA as it may have altered expected feature access rights for existing supervisor/caseworkers - specs suggest not.

There is a new routing table in `application_controller` that routes users after sign_in based on their role, rather than defaulting to the 'manage users' page.

Similarly, a new `ForbiddenError` has been introduced which takes the place of the default routing to the 'Manage Users' page.

The primary navigation showing open applications, all applications, etc links has been restricted to 'service users' only.

## Screenshots of changes (if applicable)

### Before changes:
N/A

### After changes:
#### New ForbiddenError results in access restricted message
<img width="781" alt="Screenshot 2023-09-08 at 08 24 31" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/47113046/63b3df67-e8b0-4092-8426-aa1ad93779d0">


#### Down migration (rollback) is no longer possible
<img width="603" alt="Screenshot 2023-09-08 at 00 25 12" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/47113046/233eb4e8-61bd-4663-a703-cb51aaeab2b4">



## How to manually test the feature
1. Login as a caseworker - should should show all sections as expected
2. Login as supervisor - should show all sections as expected
